test/CMakeLists.txt

```diff
@@ -72,33 +100,47 @@ set(test_dsn_counter 0)
 foreach (test_dsn ${TEST_DSN_LIST})
     if (test_dsn)
         if(PERL AND (
-                ("${ODBC_PROVIDER}" STREQUAL "MDAC" AND PERL_USES_MDAC) OR
-                ("${ODBC_PROVIDER}" STREQUAL "UnixODBC" AND PERL_USES_UNIXODBC) OR
-                ("${ODBC_PROVIDER}" STREQUAL "iODBC" AND PERL_USES_IODBC)
+            ("${ODBC_PROVIDER}" STREQUAL "MDAC"     AND PERL_USES_MDAC) OR
+            ("${ODBC_PROVIDER}" STREQUAL "UnixODBC" AND PERL_USES_UNIXODBC) OR
+            ("${ODBC_PROVIDER}" STREQUAL "iODBC"    AND PERL_USES_IODBC)
         ))
-            add_test(NAME "test.pl-a-dsn-${test_dsn_counter}" COMMAND ${PERL} ${CMAKE_CURRENT_SOURCE_DIR}/test.pl "${test_dsn}" a)
-            add_test(NAME "test.pl-w-dsn-${test_dsn_counter}" COMMAND ${PERL} ${CMAKE_CURRENT_SOURCE_DIR}/test.pl "${test_dsn}" w)
+            add_test(NAME "test.pl-dsn-${test_dsn_counter}" COMMAND ${PERL} ${CMAKE_CURRENT_SOURCE_DIR}/test.pl "${test_dsn}")
```

Comment:
The only possible extension to that naming would be just putting the full DSN there, since Unicode vs ANSI driver type could not be deduced at this point. That won't be good for the readability/usability of tests, might even not be possible syntactically. The ODBC provider in its turn is fixed for the whole build, so no need to mention it there.

Reply:
```
--- test/CMakeLists.txt
+++ test/CMakeLists.txt
@@ -98,13 +98,15 @@ set(SQL_ANSWER "580245" CACHE STRING "")
 
 set(test_dsn_counter 0)
 foreach (test_dsn ${TEST_DSN_LIST})
+    string(REGEX MATCH ".*\\(([a-zA-Z]+)\\)" _ "${test_dsn}")
+    set(TEST_SUFFIX "${CMAKE_MATCH_1}_${ODBC_PROVIDER}")
     if (test_dsn)
         if(PERL AND (
             ("${ODBC_PROVIDER}" STREQUAL "MDAC"     AND PERL_USES_MDAC) OR
             ("${ODBC_PROVIDER}" STREQUAL "UnixODBC" AND PERL_USES_UNIXODBC) OR
             ("${ODBC_PROVIDER}" STREQUAL "iODBC"    AND PERL_USES_IODBC)
         ))
-            add_test(NAME "test.pl-dsn-${test_dsn_counter}" COMMAND ${PERL} ${CMAKE_CURRENT_SOURCE_DIR}/test.pl "${test_dsn}")
+            add_test(NAME "test.pl-dsn-${TEST_SUFFIX}" COMMAND ${PERL} ${CMAKE_CURRENT_SOURCE_DIR}/test.pl "${test_dsn}")
         endif()
 
         if(PYTHON2 AND (
```
etc, results in:
```
$ ctest -N
Test project /home/enmk/proj/.builds/clickhouse-odbc-GCC_9-Debug
  Test  #1: clickhouse-odbc-ut
  Test  #2: clickhouse-odbc-load-ut
  Test  #3: clickhouse-odbc-load-ut-w
  Test  #4: clickhouse-odbc-client-it-dsn-0
  Test  #5: clickhouse-odbc-client-it-dsn-1
  Test  #6: clickhouse-odbc-nano-it-dsn-0
  Test  #7: clickhouse-odbc-nano-it-dsn-1
  Test  #8: clickhouse-odbcw-ut
  Test  #9: clickhouse-odbcw-load-ut
  Test #10: clickhouse-odbcw-load-ut-w
  Test #11: clickhouse-odbcw-client-it-dsn-0
  Test #12: clickhouse-odbcw-client-it-dsn-1
  Test #13: clickhouse-odbcw-nano-it-dsn-0
  Test #14: clickhouse-odbcw-nano-it-dsn-1
  Test #15: isql-dsn-ANSI_UnixODBC
  Test #16: test.sh-isql-dsn-ANSI_UnixODBC
  Test #17: iusql-dsn-ANSI_UnixODBC
  Test #18: test.sh-iusql-dsn-ANSI_UnixODBC
  Test #19: isql-dsn-Unicode_UnixODBC
  Test #20: test.sh-isql-dsn-Unicode_UnixODBC
  Test #21: iusql-dsn-Unicode_UnixODBC
  Test #22: test.sh-iusql-dsn-Unicode_UnixODBC

Total Tests: 22
```

Reply:
That's not reliable, DSNs don't have to follow that naming pattern in general. Also, doesn't look too readable.